### PR TITLE
Develop mg

### PR DESCRIPTION
**Pull request by mngauthier**
Three options of photosynthesis model are proposed :
- run the initial photosynthesis sub-model (as in Barillot 2016 and in Gauthier 2020)
- regulation by specific proteins instead of SLN
- regulation by specific proteins instead of SLN + feedback inhibition by non-structural carbohydrates

A parameter file was created.

### CODE REVIEW

File: farquharwheat/converter.py [hunk 1]
```diff
@@ -16,12 +16,13 @@
...
 #: the inputs needed by FarquharWheat at element scale
-FARQUHARWHEAT_ELEMENTS_INPUTS = ['width', 'height', 'PARa', 'nitrates', 'amino_acids', 'proteins', 'Nstruct', 'green_area']
+FARQUHARWHEAT_ELEMENTS_INPUTS = ['width', 'height', 'PARa', 'nitrates', 'amino_acids', 'proteins', 'Nstruct', 'green_area',
+                                 'sucrose', 'starch', 'fructan']
 #: the inputs needed by FarquharWheat at axis scale
 FARQUHARWHEAT_AXES_INPUTS = ['SAM_temperature', 'height_canopy']
 
 #: the outputs computed by FarquharWheat
-FARQUHARWHEAT_ELEMENTS_OUTPUTS = ['Ag', 'An', 'Rd', 'Tr', 'Ts', 'gs', 'width', 'height']
+FARQUHARWHEAT_ELEMENTS_OUTPUTS = ['Ag', 'An', 'Rd', 'Ac', 'Aj', 'Ap', 'Ag_before_inhibition_WSC', 'Tr', 'Ts', 'gs', 'width', 'height']
```
Comment (rbarillot) [inline farquharwheat/converter.py:25]: Is it still necessary to add these outputs?

File: farquharwheat/model.py [hunk 2]
```diff
@@ -17,372 +18,371 @@
...
+    return p
+
+
+def _inhibition_by_WSC(WSC):
+    """
+    Calculates the relative diminution of Ap (or Ag) due to inhibition by WSC. Adapted from Azcon-Bieto 1983
+
+    :param float WSC: Surfacic content of water soluble carbohydrates  (�mol C m-2)
+
+    :return: Ci (�mol mol-1)
```
Comment (rbarillot) [inline farquharwheat/model.py:172]: You sure about the name of the output variable 'Ci'?

File: farquharwheat/model.py [hunk 3]
```diff
@@ -17,372 +18,371 @@
...
+    # I think there was a mistake in the paper of Braune t al. (2009) where they wrote Ap = (1-Gamma/Ci)*(3*TPU) + Vo
+    # A more recent expression of Ap was given by S. v Caemmerer in her book (2000): AP = (3TPU * (Ci-Gamma))/(Ci-(1+3alpha)*Gamma),
+    # where 0 < alpha > 1 is the fraction of glycolate carbon not returned to the chloroplast, but I couldn't find any estimation of alpha for wheat
+
+    #: Gross assimilation rate (�mol m-2 s-1)
+    if option_Retroinhibition:
+        Ag_before_inhibition_WSC = min(Ac, Aj)
+        Ag = Ag_before_inhibition_WSC * (1 - _inhibition_by_WSC(surfacic_WSC))
+    else:
+        Ag_before_inhibition_WSC = Ag = min(Ac, Aj, Ap)
```
Comment (rbarillot) [inline farquharwheat/model.py:238]: still useful to have Ag_before_inhibition_WSC? Could be called 'Ag'

File: farquharwheat/simulation.py [hunk 4]
```diff
@@ -84,22 +89,54 @@ def run(self, Ta, ambient_CO2, RH, Ur):
                 continue
             # In case it is an HiddenElement, we need temperature calculation. Cases of Visible Element without geomtry proprety (because too small) don't have photosynthesis calculation neither.
             if element_inputs['height'] is None:
-                Ag, An, Rd, Tr, gs = 0.0, 0.0, 0.0, 0.0, 0.0
+                Ag, An, Rd, Tr, gs = 0., 0., 0., 0., 0.
+                Ac, Aj, Ap, Ag_before_inhibition_WSC = 0., 0., 0., 0.
```
Comment (rbarillot) [inline farquharwheat/simulation.py:93]: 'Ag_before_inhibition_WSC ' useful?

**Review by rbarillot — CHANGES_REQUESTED** [review]
Could you also change the version number in the __init__.py file. It could be 2.0

File: farquharwheat/simulation.py (hunk 4 above)
Comment (mngauthier) [reply farquharwheat/simulation.py:93]: I believe 'Ag_before_inhibition_WSC ' is a usefull output to know whether retro-inhibition by sugar limited photosynthesis.

File: farquharwheat/model.py (hunk 3 above)
Comment (mngauthier) [reply farquharwheat/model.py:238]: I believe 'Ag_before_inhibition_WSC ' is a usefull output to know whether retro-inhibition by sugar limited photosynthesis.

File: farquharwheat/simulation.py (hunk 4 above)
Comment (mngauthier) [reply farquharwheat/simulation.py:93]: I have deleted the intermediate variable 'Ag_before_inhibition_WSC'

File: farquharwheat/converter.py (hunk 1 above)
Comment (mngauthier) [reply farquharwheat/converter.py]: I deleted unused variables.

**Comment by mngauthier** [comment]
I made the changes.

**Review by rbarillot — APPROVED** [review]

**Pull request merged**